phac-nml/irida-next
# Concatenation Modal - SessionStorage clear fix

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR fixes 2 bugs in the concatenation UI functionality:

1) Providing a basename which doesn't match the pattern set for the basename input in the concatenation modal then fixing it and trying to concatenate again presents the user with an error
2) Selecting 2 pairs of paired-end files, deleting one pair, then clicking the `Concatenate` and trying to concatenate presents the user with an error

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

There are 2 scenarios to test broken functionality (on main branch). **Note: Once these are tested against the `main` branch verify that the issues have been fixed with this fix PR.**

### Concatenation failing with no files selected
1) Login as a user, create a project, a sample, and upload a few files to the sample
2) Select the files and de-select them
3) Create another sample in the project and upload files
4) Select these files
5) In the browser tools check that the sessionstorage now has an incorrect storage key (the url in the browser and this key are different)
6) Click the `Concatenate Files` button. In the modal provide a basename with a space in the text, and click the `Concatenate` button
7) Verify that an HTML5 validation error is displayed in a tooltip for the input as to the pattern not matching
8) Update the basename to include no spaces and click the `Concatenate` button once again
9) Verify that you see an error that no files are selected

### Concatenation button not disabled when 2 pairs of paired-end files are selected and one of them is deleted
1) Login as a user, create a project, a sample, and upload 2 pairs of paired-end files
2) Select the 2 files and then delete one of the pairs
3) In the browser tools check that the sessionstorage still has both pairs saved
4) Verify that the `Concatenate Files` button is still enabled
5) Click the `Concatenate Files` button
6) Verify in the modal you only see one pair of files
7) Provide a basename and click the `Concatenate` button
8) Verify that an error is present that not all or some of the selected files do not belong to the sample


### Concatenation failing

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
